tests/unit/test_tap_kafka.py

```diff
@@ -1016,46 +946,46 @@ def test_iso_timestamp_to_epoch(self):
 
-    @patch('tap_kafka.sync.assign_consumer_to_bookmarked_state')
-    @patch('tap_kafka.sync.assign_consumer_to_timestamp')
-    def test_assign_consumer(self, assign_consumer_to_timestamp, assign_consumer_to_bookmarked_state):
-        consumer = KafkaConsumerMock([])
-
-        # Should not assign if both state and initial_start_time are empty
-        sync.assign_consumer(consumer, topic='test-topic', state={}, initial_start_time=None)
-        self.assertEqual(assign_consumer_to_timestamp.call_count, 0)
-        self.assertEqual(assign_consumer_to_bookmarked_state.call_count, 0)
-
-        # Should not assign if state is empty and initial_start_time is the reserved 'latest'
-        sync.assign_consumer(consumer, topic='test-topic', state={}, initial_start_time='latest')
-        self.assertEqual(assign_consumer_to_timestamp.call_count, 0)
-        self.assertEqual(assign_consumer_to_bookmarked_state.call_count, 0)
-
-        # Should not assign if state is empty and initial_start_time is the reserved 'earliest'
-        sync.assign_consumer(consumer, topic='test-topic', state={}, initial_start_time='earliest')
-        self.assertEqual(assign_consumer_to_timestamp.call_count, 0)
-        self.assertEqual(assign_consumer_to_bookmarked_state.call_count, 0)
-
-        # Should assign by timestamp if state not provided and initial_start_time is an ISO 8601 timestamp
-        sync.assign_consumer(consumer, topic='test-topic', state={}, initial_start_time='2021-11-01 12:00:00')
-        self.assertEqual(assign_consumer_to_timestamp.call_count, 1)
-        self.assertEqual(assign_consumer_to_bookmarked_state.call_count, 0)
-
-        # Should assign by bookmark if bookmark provided
-        sync.assign_consumer(consumer, topic='test-topic', state={'bmrk': []}, initial_start_time=None)
-        self.assertEqual(assign_consumer_to_timestamp.call_count, 1)
-        self.assertEqual(assign_consumer_to_bookmarked_state.call_count, 1)
-
-        # Should assign by bookmark if both state and initial_start_time are provided
-        sync.assign_consumer(consumer, topic='test-topic', state={'bmrk': []}, initial_start_time='2021-11-01 12:00:00')
-        self.assertEqual(assign_consumer_to_timestamp.call_count, 1)
-        self.assertEqual(assign_consumer_to_bookmarked_state.call_count, 2)
-
-    def test_seek_partitions(self):
-        # TODO: This is just a fake unit test to reach the coverage limit, please complete it!
-        consumer = KafkaConsumerMock([])
-        sync.seek_partitions(consumer, [])
+    def test_epoch_to_iso_timestamp(self):
+        """Validate converting epoch milliseconds to ISO timestamp"""
+        self.assertEqual(sync.epoch_to_iso_timestamp(1635807671000),'2021-11-01T23:01:11.000')
+        self.assertEqual(sync.epoch_to_iso_timestamp(1635807671123),'2021-11-01T23:01:11.123')
+
+        # Invalid epoch should raise exception
+        with self.assertRaises(InvalidTimestampException):
```

Comment:
Based on the testing invalid values, I can't understand what we expect from the method if input value be `None`!
This test can be passed yet if the method be updated to return a value if input value be `None`
for example if I add this line at the beginning of the method:
```python
    if epoch is None:
        return datetime.datetime.utcfromtimestamp(1000000000)
```
_A good test should fail if the under test method is manipulated (by mistake, or ...etc)_

Reply:
Added another test to fail if None is passed as a supposed epoch timestamp